### ISSUE TITLE
Make sure that we only retry bundles 3 times

### ISSUE THREAD

**Issue by wezell** [dotCMS/core#21579]
We have a few cases where a bundle will continue to retry bundles indefinitely.  This might be for various causes, db inconsistency, We need to ensure that we only retry bundles 3 times and then fail them.  If the audit log data gets out of whack, we should either rewrite it or delete the remnant bundle from the db.

```
[21/01/22 16:13:15:610 UTC] WARN web.ContentletWebAPIImpl: com.dotcms.util.ThreadContextUtil.wrapVoidNoReindex(ThreadContextUtil.java:88)
[21/01/22 16:14:00:057 UTC] INFO business.PublisherQueueJob: ===========================================================
[21/01/22 16:14:00:057 UTC] INFO business.PublisherQueueJob: For bundle '01FFG4WM1CH95C6YZEW23TV57X':
[21/01/22 16:14:00:057 UTC] INFO business.PublisherQueueJob: -> Status : FAILED_TO_SEND_TO_ALL_GROUPS [4]
[21/01/22 16:14:00:057 UTC] INFO business.PublisherQueueJob: -> Re-publish attempts: 2 out of 3
[21/01/22 16:14:00:057 UTC] INFO business.PublisherQueueJob: -> NOTE : This bundle is stalled and its Audit History will NOT be updated properly. You can safely ignore this status, or you can delete its Audit History directly from the database.
[21/01/22 16:14:00:057 UTC] INFO business.PublisherQueueJob: ===========================================================
[21/01/22 16:15:00:073 UTC] INFO business.PublisherQueueJob: ===========================================================
[21/01/22 16:15:00:073 UTC] INFO business.PublisherQueueJob: For bundle '01FFG4WM1CH95C6YZEW23TV57X':
[21/01/22 16:15:00:073 UTC] INFO business.PublisherQueueJob: -> Status : FAILED_TO_SEND_TO_ALL_GROUPS [4]
[21/01/22 16:15:00:073 UTC] INFO business.PublisherQueueJob: -> Re-publish attempts: 2 out of 3
[21/01/22 16:15:00:073 UTC] INFO business.PublisherQueueJob: -> NOTE : This bundle is stalled and its Audit History will NOT be updated properly. You can safely ignore this status, or you can delete its Audit History directly from the database.
[21/01/22 16:15:00:073 UTC] INFO business.PublisherQueueJob: ===========================================================
[21/01/22 16:15:16:700 UTC] INFO taillog.TailLogServlet: Requested logFile:/srv/dotserver/tomcat-9.0.41/logs/dotcms.log
```

discussion: https://github.com/dotCMS/core/discussions/21592

**Comment by freddyucv** [comment]
Hey team! Please [add your planning poker estimate](https://app.zenhub.com/workspaces/Falcon-6176e9d907eb0000106ee8e1/issues/dotCMS/core/21579?planning-poker) with ZenHub @dsilvam @victoralfaro-dotcms

**Comment by wezell** [comment]
something to consider for the next LTS > 21.06

**Comment by dsilvam** [comment]
Note to QA.

Original way to reproduce the issue was:

* Configure an invalid endpoint (bad url or port or token)
* Push any asset, can be a piece of content
* After the first failure go to Site->Publishing Queue->Pending tab and delete the asset (not the bundle) 

This would make the retry to go indefinitely. Now instead it should not retry anymore

**Comment by victoralfaro-dotcms** [comment]
Environment
Sender: http://localhost:8080
Receiver: http://localhost:8081
Configured receiver endpoint with wrong port.
<img width="337" alt="image" src="https://user-images.githubusercontent.com/64026161/155768744-47e22e0e-20a9-494a-b92f-47fc4ca71b63.png">
Attempts:
```
12:33:00.214  INFO  util.PushPublishLogger - class com.dotcms.publisher.business.PublisherQueueJob : Status update: Failed to update bundle audit status.
12:33:00.258  INFO  business.PublisherQueueJob - ===========================================================
12:33:00.259  INFO  business.PublisherQueueJob - For bundle '01FWS2G3FQ24Y0XBDF6KPESBHN':
12:33:00.259  INFO  business.PublisherQueueJob - -> Status             : FAILED_TO_SEND_TO_ALL_GROUPS [4]
12:33:00.259  INFO  business.PublisherQueueJob - -> Re-publish attempts: 1 out of 3
12:33:00.259  INFO  business.PublisherQueueJob - ===========================================================
12:33:00.259  INFO  business.PublisherQueueJob -
12:33:00.259  INFO  business.PublisherQueueJob - Found 1 bundle(s) to process.
12:33:00.259  INFO  business.PublisherQueueJob -
12:33:00.259  INFO  business.PublisherQueueJob - Processing bundle: ID: 01FWS2G3FQ24Y0XBDF6KPESBHN. Status: 4. Publish Date: 2022-02-25 18:32:00.0
```
```
12:34:00.098  INFO  util.PushPublishLogger - class com.dotcms.publisher.business.PublisherQueueJob : Status update: Failed to update bundle audit status.
12:34:00.100  INFO  business.PublisherQueueJob - ===========================================================
12:34:00.100  INFO  business.PublisherQueueJob - For bundle '01FWS2G3FQ24Y0XBDF6KPESBHN':
12:34:00.101  INFO  business.PublisherQueueJob - -> Status             : FAILED_TO_SEND_TO_ALL_GROUPS [4]
12:34:00.101  INFO  business.PublisherQueueJob - -> Re-publish attempts: 2 out of 3
12:34:00.101  INFO  business.PublisherQueueJob - ===========================================================
12:34:00.101  INFO  business.PublisherQueueJob -
12:34:00.101  INFO  business.PublisherQueueJob - Found 1 bundle(s) to process.
12:34:00.101  INFO  business.PublisherQueueJob -
12:34:00.101  INFO  business.PublisherQueueJob - Processing bundle: ID: 01FWS2G3FQ24Y0XBDF6KPESBHN. Status: 4. Publish Date: 2022-02-25 18:32:00.0
```
```
12:35:00.396  INFO  business.PublishAuditAPIImpl - Deleting the bundle: 01FWS2G3FQ24Y0XBDF6KPESBHN
12:35:00.413  INFO  business.PublisherQueueJob - ===========================================================
12:35:00.413  INFO  business.PublisherQueueJob - For bundle '01FWS2G3FQ24Y0XBDF6KPESBHN':
12:35:00.413  INFO  business.PublisherQueueJob - -> Status             : FAILED_TO_PUBLISH [8]
12:35:00.413  INFO  business.PublisherQueueJob - -> Re-publish attempts: 3 out of 3
12:35:00.413  INFO  business.PublisherQueueJob - ===========================================================
12:35:00.528  INFO  business.PublisherQueueJob -
12:35:00.528  INFO  business.PublisherQueueJob - Found 1 bundle(s) to process.
12:35:00.528  INFO  business.PublisherQueueJob -
12:35:00.528  INFO  business.PublisherQueueJob - Processing bundle: ID: 01FWS2G3FQ24Y0XBDF6KPESBHN. Status: 4. Publish Date: 2022-02-25 18:32:00.0
```

**Comment by freddyucv** [comment]
Pass internal QA: Tested:

Case 1:

Create an end point with an invalid IP.
Send anything to that End point
It should retry just 3 times and then stop (check the log file)
Case 2:

Create an end point with an invalid token and a valid IP.
Send anything to that End point
It should retry just 3 times and then stop (check the log file)
Case 3:

Create an end point with an invalid IP.
Send anything to that End point
After the first fail, go to the "Pending" tab and delete the asset from the bundle (not delete the bundle)
it should not retry anymore and delete the bundle for the "Status/History" tab.
Case 4:

Create an end point with a valid IP and token.
Send anything to that End point
After send the bundle successfully to the receiver the bundle should keep in the sender "Status/History" tab
Sometimes dotCMS try to retry a bundle 4 times, but this issue is happening before this change, maybe is a concurrency bug.

**Comment by bryanboza** [comment]
Fixed, tested on release-22.03 // Docker // FF
